test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp

```diff
@@ -3,8 +3,11 @@
 
 #include <stan/math/fwd/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
 
 static inline void expect_val_deriv_eq(double exp_var, double test_var) {
+  using stan::math::is_nan;
+  if (is_nan(exp_var) && is_nan(test_var)) return;
```

Comment:
I believe this is just a way to make sure if the correct
answer is NaN that the result is NaN.  For instance,
we need to make sure log(-1) returns NaN and as far as I know,
this is the only way to do it other than writing a more general
equality function.

- Bob


> On Apr 27, 2016, at 6:21 PM, Daniel Lee <notifications@github.com> wrote:
> 
> In test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp:
> 
> >  
> >  static inline void expect_val_deriv_eq(double exp_var, double test_var) {
> > +  using stan::math::is_nan;
> > +  if (is_nan(exp_var) && is_nan(test_var)) return;
> 
> hi @rayleigh, we need to do something about NaNs. Was that a suggestion?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>